src/imageops/filter_1d.rs

```diff
@@ -0,0 +1,879 @@
+            *dst = *src;
+        }
+        return Ok(());
+    }
+
+    let pad_w = (scanned_row_kernel.len() / 2).max(1);
+
+    let arena_width = image_size.width * N + pad_w * 2 * N;
+    let mut row_buffer = vec![T::default(); arena_width];
+
+    // If column kernel is small better to use ring queue circular convolution
+
+    let full_width = image_size.width * N;
+
+    // Flat ring queue
+    let mut buffer = vec![T::default(); image_size.width * N * scanned_column_kernel.len()];
+
+    let column_kernel_len = scanned_column_kernel.len();
+
+    let half_kernel = column_kernel_len / 2;
+
+    make_arena_row::<T, N>(
+        image,
+        &mut row_buffer,
+        0,
+        image_size,
+        KernelShape {
+            width: scanned_row_kernel.len(),
+            height: 0,
+        },
+    );
+    filter_symmetric_row::<T, I, N>(&row_buffer, &mut buffer[..full_width], &scanned_row_kernel);
+
+    let (src_row, rest) = buffer.split_at_mut(full_width);
+    for dst in rest.chunks_exact_mut(full_width).take(half_kernel) {
+        for (dst, src) in dst.iter_mut().zip(src_row.iter()) {
+            *dst = *src;
+        }
+    }
+
+    let mut start_ky = column_kernel_len / 2 + 1;
+
+    start_ky %= column_kernel_len;
+
+    for y in 1..image_size.height + half_kernel {
+        let new_y = if y < image_size.height {
+            y
+        } else {
+            y.min(image_size.height - 1)
+        };
+
+        make_arena_row::<T, N>(
+            image,
+            &mut row_buffer,
+            new_y,
+            image_size,
+            KernelShape {
+                width: scanned_row_kernel.len(),
+                height: 0,
+            },
+        );
+
+        filter_symmetric_row::<T, I, N>(
+            &row_buffer,
+            &mut buffer[start_ky * full_width..(start_ky + 1) * full_width],
+            &scanned_row_kernel,
+        );
+
+        if y >= half_kernel {
+            let mut brows = vec![image; column_kernel_len];
+
+            for (i, brow) in brows.iter_mut().enumerate() {
+                let ky = (i + start_ky + 1) % column_kernel_len;
+                *brow = &buffer[ky * full_width..(ky + 1) * full_width];
+            }
+
+            let dy = y - half_kernel;
+
+            let dst = &mut destination[dy * full_width..(dy + 1) * full_width];
+
+            filter_symmetric_column::<T, I, N>(&brows, dst, image_size, &scanned_column_kernel);
+        }
+
+        start_ky += 1;
+        start_ky %= column_kernel_len;
+    }
+
+    Ok(())
+}
+
+/// Performs 2D separable convolution on the image
+///
+/// Currently implemented only for symmetrical filters
+///
+/// # Arguments
+///
+/// * `image`: Single plane image
+/// * `destination`: Destination image
+/// * `image_size`: Image size see [FilterImageSize]
+/// * `row_kernel`: Row kernel, *size must be odd*!
+/// * `column_kernel`: Column kernel, *size must be odd*!
+fn filter_1d<T, F, I, const N: usize>(
```

Comment:
I don't like `1d` in the name here, makes me expect _one_ filter and a choice of direction rather than what we have: `filter_separable` may be more appropriate. Applies to the exported versions below.